## NB: when maintaining this patch, remember that `git diff` has to be run in all of the VST3SDK submodules!

diff --git a/cmake/modules/SMTG_DetectPlatform.cmake b/cmake/modules/SMTG_DetectPlatform.cmake
index dc879fe..891dcbd 100644
--- a/cmake/modules/SMTG_DetectPlatform.cmake
+++ b/cmake/modules/SMTG_DetectPlatform.cmake
@@ -52,7 +52,7 @@ macro(smtg_detect_xcode_version)
             string(REGEX REPLACE "Xcode ([0-9\\.]+)" "\\1" XCODE_VERSION "${XCODE_VERSION}")
         endif()
         if(XCODE_VERSION VERSION_LESS "9")
-            message(FATAL_ERROR "[SMTG] XCode 9 or newer is required")
+            message(WARNING "[SMTG] XCode 9 or newer are supported only")
         endif()
         message(STATUS "[SMTG] Building with Xcode version: ${XCODE_VERSION}")
         if(NOT DEFINED ENV{SDKROOT})
diff --git a/cmake/modules/SMTG_PlatformToolset.cmake b/cmake/modules/SMTG_PlatformToolset.cmake
index 49ff310..9df779f 100644
--- a/cmake/modules/SMTG_PlatformToolset.cmake
+++ b/cmake/modules/SMTG_PlatformToolset.cmake
@@ -13,9 +13,9 @@ macro(smtg_setup_platform_toolset)
 
     # Add colors to clang output when using Ninja
     # See: https://github.com/ninja-build/ninja/wiki/FAQ
-    if(UNIX AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_GENERATOR STREQUAL "Ninja")
-        add_compile_options(-fcolor-diagnostics)
-    endif()
+#     if(UNIX AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND CMAKE_GENERATOR STREQUAL "Ninja")
+#         add_compile_options(-fcolor-diagnostics)
+#     endif()
 
     #------------
     option(SMTG_ENABLE_ADDRESS_SANITIZER "Enable Address Sanitizer" OFF)
diff --git a/vstgui4/vstgui//lib/CMakeLists.txt b/vstgui4/vstgui//lib/CMakeLists.txt
index 7d13703c..fe3fc736 100644
--- a/vstgui4/vstgui//lib/CMakeLists.txt
+++ b/vstgui4/vstgui//lib/CMakeLists.txt
@@ -341,7 +341,6 @@ if(LINUX)
 endif()
 
 if(CMAKE_HOST_APPLE)
-  target_compile_options(${target} PRIVATE -Wall -Werror)
   set(PLATFORM_LIBRARIES
 	"-framework Cocoa"
 	"-framework QuartzCore"
diff --git a/vstgui4/vstgui//lib/platform/mac/cocoa/cocoahelpers.mm b/vstgui4/vstgui//lib/platform/mac/cocoa/cocoahelpers.mm
index 8905a992..4669b70a 100644
--- a/vstgui4/vstgui//lib/platform/mac/cocoa/cocoahelpers.mm
+++ b/vstgui4/vstgui//lib/platform/mac/cocoa/cocoahelpers.mm
@@ -19,9 +19,9 @@ using namespace VSTGUI;
 //------------------------------------------------------------------------------------
 HIDDEN bool CreateKeyboardEventFromNSEvent (NSEvent* theEvent, KeyboardEvent& event)
 {
-	if (theEvent.type == NSEventTypeKeyUp)
+	if (theEvent.type == NSKeyUp)
 		event.type = EventType::KeyUp;
-	else if (theEvent.type == NSEventTypeKeyDown || theEvent.type == NSEventTypeFlagsChanged)
+	else if (theEvent.type == NSKeyDown || theEvent.type == NSFlagsChanged)
 	{
 		event.type = EventType::KeyDown;
 		if (theEvent.ARepeat)
diff --git a/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewdraggingsession.mm b/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewdraggingsession.mm
index 6ab7ab43..127094c7 100644
--- a/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewdraggingsession.mm
+++ b/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewdraggingsession.mm
@@ -63,7 +63,11 @@ struct BinaryDataType : RuntimeObjCClass<BinaryDataType>
 		obj.callSuper<void (id, SEL)> (_cmd);
 	}
 
+#if defined(MAC_OS_X_VERSION_10_11) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11)
 	static NSArray<NSPasteboardType>* WritableTypesForPasteboard (id, SEL, NSPasteboard*)
+#else
+	static NSArray* WritableTypesForPasteboard (id, SEL, NSPasteboard*)
+#endif
 	{
 		return @[getCocoaPasteboardTypeString ()];
 	}
@@ -199,7 +203,11 @@ bool NSViewDraggingSession::setBitmap (const SharedPointer<CBitmap>& bitmap, CPo
 	[session enumerateDraggingItemsWithOptions:0
 	                                   forView:nil
 	                                   classes:[NSArray arrayWithObject:[NSPasteboardItem class]]
+#if defined(MAC_OS_X_VERSION_10_11) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11)
 	                             searchOptions:[NSDictionary<NSPasteboardReadingOptionKey, id> new]
+#else
+	                             searchOptions:[NSDictionary new]
+#endif
 	                                usingBlock:[&] (NSDraggingItem* _Nonnull draggingItem,
 	                                                NSInteger idx, BOOL* _Nonnull stop) {
 		                                if (idx != 0)
diff --git a/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewframe.mm b/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewframe.mm
index bca36f48..5b98593a 100644
--- a/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewframe.mm
+++ b/vstgui4/vstgui//lib/platform/mac/cocoa/nsviewframe.mm
@@ -372,7 +372,7 @@ struct VSTGUI_NSView : RuntimeObjCClass<VSTGUI_NSView>
 	static void windowDidChangeBackingProperties (id self, SEL _cmd, NSNotification* notification)
 	{
 		double scaleFactor = 1.;
-		if (auto window = [self window])
+		if (id window = [self window])
 			scaleFactor = [window backingScaleFactor];
 
 		NSViewFrame* viewFrame = getNSViewFrame (self);
@@ -451,6 +451,7 @@ struct VSTGUI_NSView : RuntimeObjCClass<VSTGUI_NSView>
 	//------------------------------------------------------------------------
 	static void viewWillDraw (id self, SEL _cmd)
 	{
+#if defined(MAC_OS_X_VERSION_10_12) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12)
 		if (@available (macOS 10.12, *))
 		{
 			if (auto frame = getNSViewFrame (self))
@@ -461,6 +462,7 @@ struct VSTGUI_NSView : RuntimeObjCClass<VSTGUI_NSView>
 				}
 			}
 		}
+#endif
 		makeInstance (self).callSuper<void (id, SEL)> (_cmd);
 	}
 
@@ -952,6 +954,7 @@ NSViewFrame::NSViewFrame (IPlatformFrameCallback* frame, const CRect& size, NSVi
 	auto processInfo = [NSProcessInfo processInfo];
 	if ([processInfo respondsToSelector:@selector(operatingSystemVersion)])
 	{
+#if defined(MAC_OS_X_VERSION_10_11) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11)
 		// on Mac OS X 10.11 we activate layer drawing as this fixes a few issues like that only a
 		// few parts of a window are updated permanently when scrolling or manipulating a control
 		// while other parts are only updated when the malipulation ended, or CNinePartTiledBitmap
@@ -978,6 +981,7 @@ NSViewFrame::NSViewFrame (IPlatformFrameCallback* frame, const CRect& size, NSVi
 				}
 			}
 		}
+#endif
 	}
 }
 
@@ -1703,7 +1707,7 @@ void CocoaTooltipWindow::set (NSViewFrame* nsViewFrame, const CRect& rect, const
 	}
 	auto paragrapheStyle = [[NSMutableParagraphStyle new] autorelease];
 	[paragrapheStyle setParagraphStyle:[NSParagraphStyle defaultParagraphStyle]];
-	paragrapheStyle.alignment = NSTextAlignmentCenter;
+	paragrapheStyle.alignment = NSCenterTextAlignment;
 	auto string = [NSString stringWithCString:tooltip encoding:NSUTF8StringEncoding];
 	auto attributes = [NSDictionary dictionaryWithObjectsAndKeys:[NSFont toolTipsFontOfSize:0], NSFontAttributeName, paragrapheStyle, NSParagraphStyleAttributeName, nil];
 	auto attrString = [[[NSMutableAttributedString alloc] init] autorelease];
diff --git a/vstgui4/vstgui/lib/vstguibase.h b/vstgui4/vstgui/lib/vstguibase.h
index bdc58412..f521a509 100644
--- a/vstgui4/vstgui/lib/vstguibase.h
+++ b/vstgui4/vstgui/lib/vstguibase.h
@@ -7,6 +7,7 @@
 #include <cstdlib>
 #include <cstdio>
 #include <cstring>
+#include <algorithm>
 
 //-----------------------------------------------------------------------------
 // VSTGUI Version
diff --git a/vstgui4/vstgui//standalone/CMakeLists.txt b/vstgui4/vstgui//standalone/CMakeLists.txt
index f7b00b64..b44549e9 100644
--- a/vstgui4/vstgui//standalone/CMakeLists.txt
+++ b/vstgui4/vstgui//standalone/CMakeLists.txt
@@ -137,6 +137,3 @@ if(LINUX)
     target_link_libraries(${target} PRIVATE vstgui_uidescription ${LINUX_LIBRARIES} dl ${SQLITE3_LIBRARIES})
 endif()
 
-if(CMAKE_HOST_APPLE)
-  target_compile_options(${target} PRIVATE -Wall -Werror)
-endif()
diff --git a/vstgui4/vstgui//standalone/source/platform/mac/macapplication.mm b/vstgui4/vstgui//standalone/source/platform/mac/macapplication.mm
index acbc2ec3..6b4cb177 100644
--- a/vstgui4/vstgui//standalone/source/platform/mac/macapplication.mm
+++ b/vstgui4/vstgui//standalone/source/platform/mac/macapplication.mm
@@ -31,7 +31,11 @@ static_assert (false, "Need newer clang compiler!");
 	VSTGUI::Standalone::Platform::Mac::MacPreference prefs;
 	VSTGUI::Standalone::Platform::Mac::CommonDirectories commonDirecories;
 }
+#if defined(MAC_OS_X_VERSION_10_11) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11)
 @property NSArray<NSString*>* _Nullable startupOpenFiles;
+#else
+@property NSArray* _Nullable startupOpenFiles;
+#endif
 @property BOOL hasFinishedLaunching;
 @property BOOL hasTriggeredSetupMainMenu;
 @end
@@ -464,20 +468,20 @@ static CommandWithKeyList getCommandList (const char* _Nonnull group)
 //------------------------------------------------------------------------
 - (void)updateDebugMenuItems
 {
-	NSMenuItem* debugMenu = [NSApp.mainMenu itemWithTitle:@"Debug"];
+	NSMenuItem* debugMenu = [[NSApp mainMenu] itemWithTitle:@"Debug"];
 	if (debugMenu && debugMenu.submenu)
 	{
 		if (auto item = [debugMenu.submenu itemWithTitle:@"Visualize Redraw Areas"])
 		{
 			auto state =
 			    VSTGUI::getPlatformFactory ().asMacFactory ()->enableVisualizeRedrawAreas ();
-			item.state = state ? NSControlStateValueOn : NSControlStateValueOff;
+			item.state = state ? 1 : 0;
 		}
 		if (auto item = [debugMenu.submenu itemWithTitle:@"Use Asynchronous CALayer Drawing"])
 		{
 			auto state =
 			    VSTGUI::getPlatformFactory ().asMacFactory ()->getUseAsynchronousLayerDrawing ();
-			item.state = state ? NSControlStateValueOn : NSControlStateValueOff;
+			item.state = state ? 1 : 0;
 		}
 	}
 }
@@ -675,7 +679,11 @@ static CommandWithKeyList getCommandList (const char* _Nonnull group)
 }
 
 //------------------------------------------------------------------------
+#if defined(MAC_OS_X_VERSION_10_11) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11)
 - (BOOL)openFilesInternal:(nonnull NSArray<NSString*>*)filenames
+#else
+- (BOOL)openFilesInternal:(nonnull NSArray*)filenames
+#endif
 {
 	std::vector<VSTGUI::UTF8String> paths;
 	paths.reserve (filenames.count);
@@ -687,7 +695,11 @@ static CommandWithKeyList getCommandList (const char* _Nonnull group)
 }
 
 //------------------------------------------------------------------------
+#if defined(MAC_OS_X_VERSION_10_11) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_11)
 - (void)application:(nonnull NSApplication*)sender openFiles:(nonnull NSArray<NSString*>*)filenames
+#else
+- (void)application:(nonnull NSApplication*)sender openFiles:(nonnull NSArray*)filenames
+#endif
 {
 	if (!self.hasFinishedLaunching)
 	{
diff --git a/vstgui4/vstgui//standalone/source/platform/mac/macwindow.mm b/vstgui4/vstgui//standalone/source/platform/mac/macwindow.mm
index 427b124a..0c2d7a37 100644
--- a/vstgui4/vstgui//standalone/source/platform/mac/macwindow.mm
+++ b/vstgui4/vstgui//standalone/source/platform/mac/macwindow.mm
@@ -222,7 +222,7 @@ bool Window::init (const WindowConfiguration& config, IWindowDelegate& inDelegat
 	{
 		nsWindow.title = (__bridge NSString*)titleMacStr->getCFString ();
 	}
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12 && __clang_major__ >= 9
+#if defined(MAC_OS_X_VERSION_10_12) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12) && (__clang_major__ >= 9)
 	if (@available (macOS 10.12, *))
 	{
 		if (!config.groupIdentifier.empty ())
@@ -248,7 +248,7 @@ bool Window::init (const WindowConfiguration& config, IWindowDelegate& inDelegat
 	                object:nsWindow.contentView
 	                 queue:nil
 	            usingBlock:[this] (NSNotification* _Nonnull note) {
-		            auto contentViewSize = nsWindow.contentView.frame.size;
+		            auto contentViewSize = [[nsWindow contentView] frame].size;
 		            windowDidResize ({contentViewSize.width, contentViewSize.height});
 	            }];
 
@@ -821,13 +821,27 @@ WindowPtr makeWindow (const WindowConfiguration& config, IWindowDelegate& delega
 //------------------------------------------------------------------------
 @implementation VSTGUITitlebarViewController
 
+// borrowed from Qt 5.9.8
+static NSButton *createButtonWithTitle(NSString *title)
+{
+    NSButton *button = [[NSButton alloc] initWithFrame:NSZeroRect];
+    button.title = title;
+    return button;
+}
+
 //------------------------------------------------------------------------
 - (void)loadView
 {
+#if defined(MAC_OS_X_VERSION_10_12) && (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12)
 	auto control = [NSButton buttonWithTitle:@"ⓔ" target:self action:@selector (doAction:)];
-	control.showsBorderOnlyWhileMouseInside = NO;
-	control.bordered = NO;
-	control.bezelStyle = NSBezelStyleRounded;
+#else
+	NSButton *control = createButtonWithTitle(@"ⓔ");
+	control.target = self;
+	control.action = @selector (doAction:);
+#endif
+	[control setShowsBorderOnlyWhileMouseInside:NO];
+	[control setBordered:NO];
+	[control setBezelStyle:NSRoundedBezelStyle];
 	self.view = control;
 }
 
diff --git a/vstgui4/vstgui//uidescription/CMakeLists.txt b/vstgui4/vstgui//uidescription/CMakeLists.txt
index 98414bab..995a4782 100644
--- a/vstgui4/vstgui//uidescription/CMakeLists.txt
+++ b/vstgui4/vstgui//uidescription/CMakeLists.txt
@@ -219,6 +219,3 @@ else()
     message(STATUS "VSTGUI will use the embedded Expat package!")
 endif()
 
-if(CMAKE_HOST_APPLE)
-  target_compile_options(${target} PRIVATE -Wall -Werror)
-endif()
